.travis.yml

```diff
@@ -8,13 +8,15 @@ env:
   matrix:
   - BUILDTOOL=autotools
   - BUILDTOOL=cmake
-  - BUILDTOOL=autotools_old_compilers
```

Comment:
Because the travis build script (scripts/travis_ci_build.sh) has nothing to execute when BUILDTOOL=autotools_old_compilers, therefore it's totally useless and a waste of ci resources.

I might be missing something,  but it seems that the script for this case was removed at some point in the past,  and this rule is a leftover,  so I thought it was a good time to remove it.